spec/connectors/base/advanced_snippet_against_schema_validator_spec.rb

```diff
@@ -76,6 +93,41 @@
 
         it_behaves_like 'advanced snippet is valid'
       end
+
+      context 'two fields are allowed and their key must be equal to their value' do
+        let(:constraints) {
+          [
+            ->(fields) { fields.size == 2 },
+            lambda { |fields|
+              fields.each { |field_name, field_value|
+                return false unless field_name.to_s == field_value
+              }
```

Comment:
Curious, why different lambda syntax here?

Reply:
RuboCop want this syntax for multi-line lambdas

Reply:
cc @vidok something we can improve on too :)